tests/phpunit/tests/media.php

```diff
@@ -3710,6 +3710,98 @@ function() {
 		$this->assertSame( 3, $omit_threshold );
 	}
 
+	/**
+	 * Tests that wp_get_loading_attr_default() returns the expected loading attribute value before loop.
+	 *
+	 * @ticket 58211
+	 *
+	 * @covers ::wp_get_loading_attr_default
+	 *
+	 * @dataProvider data_wp_get_loading_attr_default_before_and_no_loop
+	 *
+	 * @param string $context
+	 */
+	public function test_wp_get_loading_attr_default_before_loop( $context ) {
+		global $wp_query, $wp_the_query;
+
+		$wp_query = new WP_Query( array( 'post__in' => array( self::$post_ids['publish'] ) ) );
+		$this->reset_content_media_count();
+		$this->reset_omit_loading_attr_filter();
+
+		// Lazy if not main query.
+		$this->assertSame( 'lazy', wp_get_loading_attr_default( $context ) );
+
+		// Set current query as main query.
+		$wp_the_query = $wp_query;
+
+		// For main query return false for any instance before end of get_header.
+		$this->assertFalse( wp_get_loading_attr_default( $context ) );
+		add_action(
+			'get_header',
+			function() use ( $context ) {
+				$this->assertFalse( wp_get_loading_attr_default( $context ) );
+			}
+		);
+
+		do_action( 'get_header' );
+
+		// A test for before the loop but after get_header to check it returns the default, lazy-loading.
+		$this->assertSame( 'lazy', wp_get_loading_attr_default( $context ) );
+
+		while ( have_posts() ) {
+			the_post();
+			$this->assertFalse( wp_get_loading_attr_default( 'the_post_thumbnail' ) );
+			$this->assertSame( 'lazy', wp_get_loading_attr_default( 'wp_get_attachment_image' ) );
+		}
```

Comment:
The two assertions in these lines shouldn't be there, as they test the other parts of the `wp_get_loading_attr_default()` function, i.e. while in the loop.

We'll need to instead put assertions after the `while` loop, to assert for both contexts that now the return value is `lazy` (since we're no longer before the loop, but after the loop).